ui/app/AppLayouts/Wallet/adaptors/TokenSelectorViewAdaptor.qml

```diff
@@ -27,22 +29,65 @@ QObject {
       - currencyBalance: double (e.g. `1000.42` in user's fiat currency)
       - currencyBalanceAsString: string (e.g. "1 000,42 CZK" formatted as a string according to the user's locale)
       - balanceAsString: string (`1.42` formatted as e.g. "1,42" in user's locale)
+      - iconSource: string
     */
 
     // input API
     required property var assetsModel
+
+    // expected roles: key, name, symbol, image, communityId
+    property var plainTokensBySymbolModel // optional all tokens model, no balances
+
+    // expected roles: chainId, chainName, iconUrl
     required property var flatNetworksModel
-    required property string currentCurrency // CurrenciesStore.currentCurrency, e.g. "USD"
+
+    // CurrenciesStore.currentCurrency, e.g. "USD"
+    required property string currentCurrency
 
     // optional filter properties; empty/default values means no filtering
+    property bool showAllTokens // whether to show all tokens, or just the ones we own
     property var enabledChainIds: []
     property string accountAddress
     property bool showCommunityAssets
     property string searchString
 
     // output model
     readonly property SortFilterProxyModel outputAssetsModel: SortFilterProxyModel {
-        sourceModel: assetsObjectProxyModel
+        sourceModel: showAllTokens && !!plainTokensBySymbolModel ? concatModel : assetsObjectProxyModel
+
+        proxyRoles: [
+            FastExpressionRole {
+                name: "sectionId"
+                expression: {
+                    if (!model.currentBalance)
+                        return "section_zzz"
```

Comment:
can we rename this ti something better?

Reply:
Yeah, open to suggestions; the idea is that `section_zzz` will always sort last, after any other `section_???`

Reply:
section_last or last_section?

Reply:
No, I meant that lexicographically, the `zzz` is the last in alphabet :) so `section_aaa` < `section_zzz`